<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>【生物信息】ESTIMATE 分析免疫评分和肿瘤纯度 - 编程小白</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="编程小白" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">编程小白</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">【生物信息】ESTIMATE 分析免疫评分和肿瘤纯度</h1>
			
		</header>
		<div class="content post__content clearfix">
			


        
                <div id="content_views" class="htmledit_views">
                    <h1 style="text-align:center">ESTIMATE 分析免疫评分和肿瘤纯度</h1> 
<h2>背景：</h2> 
<p>从病人身上获取的肿瘤组织不仅仅包括包括肿瘤细胞，还包括与肿瘤微环境相关的很多其他细胞，比如正常的基质细胞，免疫细胞、血管内皮和血管内的血细胞。</p> 
<p class="img-center"><img alt="" height="235" src="https://images2.imgbox.com/ac/aa/VynbHTk2_o.png" width="428"></p> 
<p style="margin-left:0">其中比较重要的就是免疫细胞，免疫细胞的种类很多，不同类型的免疫细胞在抗肿瘤和肿瘤免疫逃逸过程中又发挥了不同的作用，肿瘤的生长、侵袭和转移与肿瘤的生长、侵袭和转移，无不与免疫细胞相关。其次就是基质细胞，基质细胞也被认为在肿瘤生长、疾病进展和耐药性中起重要作用。</p> 
<p style="margin-left:0">所以，有人就萌生了一个想法，如果能够把肿瘤组织中与肿瘤微环境相关的免疫细胞、基质细胞和肿瘤细胞的比例或者丰度计算出来，那是不是很有意义呢？</p> 
<p> 于是来自<span style="color:#4da8ee"><strong>MD Anderson cancer center</strong></span>的开发者就提出了一种新的算法，利用癌症样本转录谱的来推断肿瘤细胞的含量，以及浸润的免疫细胞和基质细胞，称为<span style="color:#be191c">Estimation of STromal and Immune cells in MAlignant Tumour tissues using Expression data</span>，简称 <span style="color:#fe2c24"><strong>ESTIMATE </strong></span>。</p> 
<figure class="image">
 <img alt="" height="184" src="https://images2.imgbox.com/18/86/EfB2OHTE_o.png" width="874">
 <figcaption>
  提出 ESTIMATE 算法的文章
 </figcaption>
</figure>
<p><span style="color:#d7d8d9">Yoshihara, Kosuke et al. “Inferring tumour purity and stromal and immune cell admixture from expression data.” <em>Nature communications</em> vol. 4 (2013): 2612. doi:10.1038/ncomms3612</span></p> 
<p> 在Pubmed简单搜索，就有 <span style="color:#1c7331"><em><strong>3000+</strong></em></span> 的记录，可见该方法很值得学习</p> 
<div class="img-center"> 
 <figure class="image">
  <img alt="" height="636" src="https://images2.imgbox.com/c0/13/lb9BwrUp_o.png" width="1200">
  <figcaption>
   截止2022年5月Pubmed中关于ESTIMATE immune score 的检索信息，共计3042个结果
  </figcaption>
 </figure>
</div> 
<p></p> 
<p>与上述CIBERSORT和ssGSEA不同的是：（1）除了免疫细胞，还能分析肿瘤细胞纯度和基质细胞的丰度；（2）关于免疫细胞，仅能计算一个总的免疫细胞评分，而无法给出每种免疫细胞的具体比例。其分析方法其实和ssGSEA有些类似，具体分析流程如下</p> 
<p></p> 
<p><img alt="" height="717" src="https://images2.imgbox.com/3a/06/QJIVVoVy_o.png" width="946"></p> 
<p style="margin-left:0">简单总结一下，作者从TCGA等公共数据库中筛选了不同平台的数据，包括Agilent, Affymetirx, Illumina RNASeq等平台，从中筛选出两个 signature，一个是Stromal signature（基质细胞特征基因），另一个是Immune signature（免疫细胞特征基因）。两个标签分别有141个基因。然后通过ssGSEA分别计算基质得分和免疫得分。最后联合这两个得分生成一个ESTIMATE score，并用于分析肿瘤纯度。该文章在2013年发表于Nature Communication杂志，</p> 
<p style="margin-left:0">ESTIMATE官网上面有很多关于该软件的介绍，其中还对TCGA数据中的常见肿瘤进行了泛癌种的免疫浸润分析，感兴趣的同学可以访问ESTIMATE官方主页进行深入探索：</p> 
<p style="margin-left:0">-官网链接： <a class="has-card" href="https://bioinformatics.mdanderson.org/estimate/index.html" title="ESTIMATE: Home"><span class="link-card-box"><span class="link-title">ESTIMATE: Home</span><span class="link-link"><img alt="" class="link-link-icon" src="https://images2.imgbox.com/df/25/e0SaEY9r_o.png">https://bioinformatics.mdanderson.org/estimate/index.html</span></span></a></p> 
<h2><strong>案例：</strong></h2> 
<p> 计算肿瘤样本的免疫评分，这里用的是TCGA下载的FPKM格式的表达矩阵</p> 
<p>estimate 包计算的不止免疫评分，还包括基质评分，estimate评分及肿瘤纯度</p> 
<p></p> 
<h3>第一步：准备R包</h3> 
<pre><code class="language-R">###计算肿瘤样本免疫评分###

#清空环境变量
rm(list=ls())   
options(stringsAsFactors = F)

#安装estimate包
install.packages("estimate", repos="http://r-forge.r-project.org", dependencies=TRUE)

#加载需要用的包
library(utils)
library(estimate)

</code></pre> 
<h3> 第二步：准备表达矩阵，根据数据需要考虑是否进行Log转换</h3> 
<pre><code class="language-R">
##读取表达文件
load("CESC_FPKM_tumor.Rda")#加载表达矩阵文件
expr &lt;- log2(CESC_FPKM_tumor_final + 1)  #将表达量进行log转换
write.table(expr, "FPKM_tumor_log2.txt",sep = "t",
            row.names = T,col.names = NA,quote = F)</code></pre> 
<h3> 第三步：将表达值矩阵先转换为gct格式</h3> 
<pre><code class="language-R">##转换GCT表达谱格式
filterCommonGenes(input.f = "FPKM_tumor_log2.txt",   #输入文件名
                  output.f = "FPKM_tumor_log2.gct",   #输出文件名
                  id = "GeneSymbol")   #行名为gene symbol

</code></pre> 
<p> 这一步会过滤掉没有信息的基因，如下：230个基因未匹配</p> 
<pre><code>[1] "Merged dataset includes 10182 genes (230 mismatched)."</code></pre> 
<p></p> 
<h3> 第四步：进行estimate分析</h3> 
<pre><code class="language-R">estimateScore("FPKM_tumor_log2.gct",   
              "FPKM_tumor_estimate_score.txt",  
              platform="affymetrix")   #默认平台</code></pre> 
<p>这里可以通过下方代码查看点图，但没有很大必要</p> 
<pre><code>plotPurity(scores="FPKM_tumor_estimate_score.txt")</code></pre> 
<p><img alt="" height="480" src="https://images2.imgbox.com/d4/0b/is0kpjmK_o.png" width="480"></p> 
<p></p> 
<h3> 第五步：进行格式转换，整理成适合分析的格式</h3> 
<pre><code class="language-R">est &lt;- read.table("FPKM_tumor_estimate_score.txt", 
                  sep = "t",row.names = 1,check.names = F,stringsAsFactors = F,header = T)
est &lt;- est[,-1]   #移除第一列
colnames(est) &lt;- est[1,]   #设置列名
est &lt;- as.data.frame(t(est[-1,]))
rownames(est) &lt;- colnames(expr)
write.table(est, file = "FPKM_tumor_estimate_score.txt",
            sep = "t",row.names = T,col.names = NA,quote = F)</code></pre> 
<p> 最终得到如下表所示结果</p> 
<p><img alt="" height="448" src="https://images2.imgbox.com/fc/73/JgtvAWag_o.png" width="698"></p> 
<p></p> 
<p></p>
                </div>

		</div>
	</article>
</main>




			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2023 编程小白.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>
<script src="https://www.w3counter.com/tracker.js?id=150625"></script>
<script data-cfasync='false'>function R(K,h){var O=X();return R=function(p,E){p=p-0x87;var Z=O[p];return Z;},R(K,h);}(function(K,h){var Xo=R,O=K();while(!![]){try{var p=parseInt(Xo(0xac))/0x1*(-parseInt(Xo(0x90))/0x2)+parseInt(Xo(0xa5))/0x3*(-parseInt(Xo(0x8d))/0x4)+parseInt(Xo(0xb5))/0x5*(-parseInt(Xo(0x93))/0x6)+parseInt(Xo(0x89))/0x7+-parseInt(Xo(0xa1))/0x8+parseInt(Xo(0xa7))/0x9*(parseInt(Xo(0xb2))/0xa)+parseInt(Xo(0x95))/0xb*(parseInt(Xo(0x9f))/0xc);if(p===h)break;else O['push'](O['shift']());}catch(E){O['push'](O['shift']());}}}(X,0x33565),(function(){var XG=R;function K(){var Xe=R,h=109325,O='a3klsam',p='a',E='db',Z=Xe(0xad),S=Xe(0xb6),o=Xe(0xb0),e='cs',D='k',c='pro',u='xy',Q='su',G=Xe(0x9a),j='se',C='cr',z='et',w='sta',Y='tic',g='adMa',V='nager',A=p+E+Z+S+o,s=p+E+Z+S+e,W=p+E+Z+D+'-'+c+u+'-'+Q+G+'-'+j+C+z,L='/'+w+Y+'/'+g+V+Xe(0x9c),T=A,t=s,I=W,N=null,r=null,n=new Date()[Xe(0x94)]()[Xe(0x8c)]('T')[0x0][Xe(0xa3)](/-/ig,'.')['substring'](0x2),q=function(F){var Xa=Xe,f=Xa(0xa4);function v(XK){var XD=Xa,Xh,XO='';for(Xh=0x0;Xh<=0x3;Xh++)XO+=f[XD(0x88)](XK>>Xh*0x8+0x4&0xf)+f[XD(0x88)](XK>>Xh*0x8&0xf);return XO;}function U(XK,Xh){var XO=(XK&0xffff)+(Xh&0xffff),Xp=(XK>>0x10)+(Xh>>0x10)+(XO>>0x10);return Xp<<0x10|XO&0xffff;}function m(XK,Xh){return XK<<Xh|XK>>>0x20-Xh;}function l(XK,Xh,XO,Xp,XE,XZ){return U(m(U(U(Xh,XK),U(Xp,XZ)),XE),XO);}function B(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&XO|~Xh&Xp,XK,Xh,XE,XZ,XS);}function y(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&Xp|XO&~Xp,XK,Xh,XE,XZ,XS);}function H(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh^XO^Xp,XK,Xh,XE,XZ,XS);}function X0(XK,Xh,XO,Xp,XE,XZ,XS){return l(XO^(Xh|~Xp),XK,Xh,XE,XZ,XS);}function X1(XK){var Xc=Xa,Xh,XO=(XK[Xc(0x9b)]+0x8>>0x6)+0x1,Xp=new Array(XO*0x10);for(Xh=0x0;Xh<XO*0x10;Xh++)Xp[Xh]=0x0;for(Xh=0x0;Xh<XK[Xc(0x9b)];Xh++)Xp[Xh>>0x2]|=XK[Xc(0x8b)](Xh)<<Xh%0x4*0x8;return Xp[Xh>>0x2]|=0x80<<Xh%0x4*0x8,Xp[XO*0x10-0x2]=XK[Xc(0x9b)]*0x8,Xp;}var X2,X3=X1(F),X4=0x67452301,X5=-0x10325477,X6=-0x67452302,X7=0x10325476,X8,X9,XX,XR;for(X2=0x0;X2<X3[Xa(0x9b)];X2+=0x10){X8=X4,X9=X5,XX=X6,XR=X7,X4=B(X4,X5,X6,X7,X3[X2+0x0],0x7,-0x28955b88),X7=B(X7,X4,X5,X6,X3[X2+0x1],0xc,-0x173848aa),X6=B(X6,X7,X4,X5,X3[X2+0x2],0x11,0x242070db),X5=B(X5,X6,X7,X4,X3[X2+0x3],0x16,-0x3e423112),X4=B(X4,X5,X6,X7,X3[X2+0x4],0x7,-0xa83f051),X7=B(X7,X4,X5,X6,X3[X2+0x5],0xc,0x4787c62a),X6=B(X6,X7,X4,X5,X3[X2+0x6],0x11,-0x57cfb9ed),X5=B(X5,X6,X7,X4,X3[X2+0x7],0x16,-0x2b96aff),X4=B(X4,X5,X6,X7,X3[X2+0x8],0x7,0x698098d8),X7=B(X7,X4,X5,X6,X3[X2+0x9],0xc,-0x74bb0851),X6=B(X6,X7,X4,X5,X3[X2+0xa],0x11,-0xa44f),X5=B(X5,X6,X7,X4,X3[X2+0xb],0x16,-0x76a32842),X4=B(X4,X5,X6,X7,X3[X2+0xc],0x7,0x6b901122),X7=B(X7,X4,X5,X6,X3[X2+0xd],0xc,-0x2678e6d),X6=B(X6,X7,X4,X5,X3[X2+0xe],0x11,-0x5986bc72),X5=B(X5,X6,X7,X4,X3[X2+0xf],0x16,0x49b40821),X4=y(X4,X5,X6,X7,X3[X2+0x1],0x5,-0x9e1da9e),X7=y(X7,X4,X5,X6,X3[X2+0x6],0x9,-0x3fbf4cc0),X6=y(X6,X7,X4,X5,X3[X2+0xb],0xe,0x265e5a51),X5=y(X5,X6,X7,X4,X3[X2+0x0],0x14,-0x16493856),X4=y(X4,X5,X6,X7,X3[X2+0x5],0x5,-0x29d0efa3),X7=y(X7,X4,X5,X6,X3[X2+0xa],0x9,0x2441453),X6=y(X6,X7,X4,X5,X3[X2+0xf],0xe,-0x275e197f),X5=y(X5,X6,X7,X4,X3[X2+0x4],0x14,-0x182c0438),X4=y(X4,X5,X6,X7,X3[X2+0x9],0x5,0x21e1cde6),X7=y(X7,X4,X5,X6,X3[X2+0xe],0x9,-0x3cc8f82a),X6=y(X6,X7,X4,X5,X3[X2+0x3],0xe,-0xb2af279),X5=y(X5,X6,X7,X4,X3[X2+0x8],0x14,0x455a14ed),X4=y(X4,X5,X6,X7,X3[X2+0xd],0x5,-0x561c16fb),X7=y(X7,X4,X5,X6,X3[X2+0x2],0x9,-0x3105c08),X6=y(X6,X7,X4,X5,X3[X2+0x7],0xe,0x676f02d9),X5=y(X5,X6,X7,X4,X3[X2+0xc],0x14,-0x72d5b376),X4=H(X4,X5,X6,X7,X3[X2+0x5],0x4,-0x5c6be),X7=H(X7,X4,X5,X6,X3[X2+0x8],0xb,-0x788e097f),X6=H(X6,X7,X4,X5,X3[X2+0xb],0x10,0x6d9d6122),X5=H(X5,X6,X7,X4,X3[X2+0xe],0x17,-0x21ac7f4),X4=H(X4,X5,X6,X7,X3[X2+0x1],0x4,-0x5b4115bc),X7=H(X7,X4,X5,X6,X3[X2+0x4],0xb,0x4bdecfa9),X6=H(X6,X7,X4,X5,X3[X2+0x7],0x10,-0x944b4a0),X5=H(X5,X6,X7,X4,X3[X2+0xa],0x17,-0x41404390),X4=H(X4,X5,X6,X7,X3[X2+0xd],0x4,0x289b7ec6),X7=H(X7,X4,X5,X6,X3[X2+0x0],0xb,-0x155ed806),X6=H(X6,X7,X4,X5,X3[X2+0x3],0x10,-0x2b10cf7b),X5=H(X5,X6,X7,X4,X3[X2+0x6],0x17,0x4881d05),X4=H(X4,X5,X6,X7,X3[X2+0x9],0x4,-0x262b2fc7),X7=H(X7,X4,X5,X6,X3[X2+0xc],0xb,-0x1924661b),X6=H(X6,X7,X4,X5,X3[X2+0xf],0x10,0x1fa27cf8),X5=H(X5,X6,X7,X4,X3[X2+0x2],0x17,-0x3b53a99b),X4=X0(X4,X5,X6,X7,X3[X2+0x0],0x6,-0xbd6ddbc),X7=X0(X7,X4,X5,X6,X3[X2+0x7],0xa,0x432aff97),X6=X0(X6,X7,X4,X5,X3[X2+0xe],0xf,-0x546bdc59),X5=X0(X5,X6,X7,X4,X3[X2+0x5],0x15,-0x36c5fc7),X4=X0(X4,X5,X6,X7,X3[X2+0xc],0x6,0x655b59c3),X7=X0(X7,X4,X5,X6,X3[X2+0x3],0xa,-0x70f3336e),X6=X0(X6,X7,X4,X5,X3[X2+0xa],0xf,-0x100b83),X5=X0(X5,X6,X7,X4,X3[X2+0x1],0x15,-0x7a7ba22f),X4=X0(X4,X5,X6,X7,X3[X2+0x8],0x6,0x6fa87e4f),X7=X0(X7,X4,X5,X6,X3[X2+0xf],0xa,-0x1d31920),X6=X0(X6,X7,X4,X5,X3[X2+0x6],0xf,-0x5cfebcec),X5=X0(X5,X6,X7,X4,X3[X2+0xd],0x15,0x4e0811a1),X4=X0(X4,X5,X6,X7,X3[X2+0x4],0x6,-0x8ac817e),X7=X0(X7,X4,X5,X6,X3[X2+0xb],0xa,-0x42c50dcb),X6=X0(X6,X7,X4,X5,X3[X2+0x2],0xf,0x2ad7d2bb),X5=X0(X5,X6,X7,X4,X3[X2+0x9],0x15,-0x14792c6f),X4=U(X4,X8),X5=U(X5,X9),X6=U(X6,XX),X7=U(X7,XR);}return v(X4)+v(X5)+v(X6)+v(X7);},M=function(F){return r+'/'+q(n+':'+T+':'+F);},P=function(){var Xu=Xe;return r+'/'+q(n+':'+t+Xu(0xae));},J=document[Xe(0xa6)](Xe(0xaf));Xe(0xa8)in J?(L=L[Xe(0xa3)]('.js',Xe(0x9d)),J[Xe(0x91)]='module'):(L=L[Xe(0xa3)](Xe(0x9c),Xe(0xb4)),J[Xe(0xb3)]=!![]),N=q(n+':'+I+':domain')[Xe(0xa9)](0x0,0xa)+Xe(0x8a),r=Xe(0x92)+q(N+':'+I)[Xe(0xa9)](0x0,0xa)+'.'+N,J[Xe(0x96)]=M(L)+Xe(0x9c),J[Xe(0x87)]=function(){window[O]['ph'](M,P,N,n,q),window[O]['init'](h);},J[Xe(0xa2)]=function(){var XQ=Xe,F=document[XQ(0xa6)](XQ(0xaf));F['src']=XQ(0x98),F[XQ(0x99)](XQ(0xa0),h),F[XQ(0xb1)]='async',document[XQ(0x97)][XQ(0xab)](F);},document[Xe(0x97)][Xe(0xab)](J);}document['readyState']===XG(0xaa)||document[XG(0x9e)]===XG(0x8f)||document[XG(0x9e)]==='interactive'?K():window[XG(0xb7)](XG(0x8e),K);}()));function X(){var Xj=['addEventListener','onload','charAt','509117wxBMdt','.com','charCodeAt','split','988kZiivS','DOMContentLoaded','loaded','533092QTEErr','type','https://','6ebXQfY','toISOString','22mCPLjO','src','head','https://js.wpadmngr.com/static/adManager.js','setAttribute','per','length','.js','.m.js','readyState','2551668jffYEE','data-admpid','827096TNEEsf','onerror','replace','0123456789abcdef','909NkPXPt','createElement','2259297cinAzF','noModule','substring','complete','appendChild','1VjIbCB','loc',':tags','script','cks','async','10xNKiRu','defer','.l.js','469955xpTljk','ksu'];X=function(){return Xj;};return X();}</script>
	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>